### DISCUSSION TITLE
Routing rules for xray not working with domain regexp or indicated directly

### DISCUSSION THREAD

**Discussion by XhonorSkillline** [XTLS/Xray-core#3621] [Q&A]
I use a server with an installed XRay proxy. The problem is that I want to configure it so that the proxy is not used for certain domains or for a domain mask. But for some reason it does not work. If you configure it through NekoBox, then everything is ok, but I want to configure it on the server so that I do not have to configure it on different clients each time. But it does not work. What am I doing wrong? Here is an example on the site https://ifconfig.me/

IP Address 185.255.111.111 But this is proxy IP.

I change config here: _/usr/local/etc/xray/config.json_

Routing settings:

```
"routing": {
    "rules":[
      {
        "type": "field",
        "ip": [
          "geoip:private",
          "geoip:ru"
        ],
        "domain": [
          "domain:vk.com",
          "domain:ifconfig.me",
          "ifconfig.me",
          "regexp:\\.ru$"
        ],
        "outboundTag": "direct"
      }
    ],
    "domainStrategy": "IPIfNonMatch"
  }
```

**Comment by kingwilliam** [comment]
There are some things that we need to reach a consensus on first

logical
```
.T. AND .T. = .T.
.T. AND .F. = .F.
.F. AND .T. = .F.
.F. AND .F. = .F.

.T. OR .T. = .T.
.T. OR .F. = .T.
.F. OR .T. = .T.
.F. OR .F. = .F. 
```

back to your config
```
1.
        "ip": [
          "geoip:private",
          "geoip:ru"
        ],
```
this is **.OR.** logical
private or "ru"



```
2.
        "domain": [
          "domain:vk.com",
          "domain:ifconfig.me",
          "ifconfig.me",
          "regexp:\\.ru$"
        ],
```
this is **.OR.** logical
"domain:vk.com" .OR. "domain:ifconfig.me"  .OR. "ifconfig.me"  .OR. "regexp:\\.ru$"

```
3.
    "rules":[
      {
        "type": "field",
        "ip": [
          "geoip:private",
          "geoip:ru"
        ],
        "domain": [
          "domain:vk.com",
          "domain:ifconfig.me",
          "ifconfig.me",
          "regexp:\\.ru$"
        ],
        "outboundTag": "direct"
      }
```
this is **.AND.** logical
"ip" .AND. "domain" BOTH NEED are TRUE logical will go "outboundTag": "direct"

final
on xtls "IP" is destination ip address, not come from ip address
if you want only accept some **incoming ip address**, you may use "source"

reference:
https://xtls.github.io/en/config/routing.html#routingobject

**Reply by ang3el7z** [reply]
I wrote my question below can you help

**Comment by ang3el7z** [comment]
I also agree with the question, this is what it looks like in x-ui, but traffic still only goes through the server
 
{
  "api": {
    "services": [
      "HandlerService",
      "LoggerService",
      "StatsService"
    ],
    "tag": "api"
  },
  "inbounds": [
    {
      "listen": "127.0.0.1",
      "port": 62789,
      "protocol": "dokodemo-door",
      "settings": {
        "address": "127.0.0.1"
      },
      "tag": "api"
    }
  ],
  "log": {
    "loglevel": "warning"
  },
  "outbounds": [
    {
      "protocol": "freedom",
      "settings": {
        "domainStrategy": "UseIP"
      },
      "tag": "direct"
    },
    {
      "protocol": "blackhole",
      "settings": {},
      "tag": "blocked"
    }
  ],
  "policy": {
    "levels": {
      "0": {
        "statsUserDownlink": true,
        "statsUserUplink": true
      }
    },
    "system": {
      "statsInboundDownlink": true,
      "statsInboundUplink": true
    }
  },
  "routing": {
    "rules": [
      {
        "type": "field",
        "ip": [
          "geoip:private",
          "geoip:ru"
        ],
        "domain": [
          "domain:vk.com",
          "domain:ifconfig.me",
          "ifconfig.me",
          "regexp:\\.ru$"
        ],
        "outboundTag": "direct"
      }
    ]
  },
  "stats": {}
}

**Comment by kingwilliam** [comment]
Can you briefly explain what result you want to achieve?

because, your routing  rules have some conflict.

**Reply by ang3el7z** [reply]
I want there to be 2 types of traffic. One foreign one went through vps, and Russian traffic, so to speak, turned around and used my ip

**Reply by kingwilliam** [reply]
what client are you use? pc / mobile ?

traffic flow ?
client -> (internet) -> vps(overseas) ->  (internet)

**Reply by ang3el7z** [reply]
1) pc and mobile |
2)
RU Trafic : client -> (internet IP MY) -> vps(overseas) -> (internet IP VPS)
Other Trafic : client -> (internet IP MY) <-> vps(overseas)

**Reply by kingwilliam** [reply]
you means

RU IP
client -> (internet RU IP) -> vps(overseas) -> (internet VPS IP) -> destination web server

others country (not ru)
client -> (internet xx IP) -> vps(overseas) -> (internet xx IP) -> destination web server
[use client ip to destination server, no use vps ip?]

**Reply by ang3el7z** [reply]
Well, just the opposite: for ru, use the client’s ip address, and for the rest, use ip from vps, I know that you can do this in the application on the client, make a direct direction for ru, and the rest through vps. but I want a universal solution, without client configuration

**Reply by kingwilliam** [reply]
because xtls is proxy service, only use vps ip to connect final server
